xds/src/main/java/io/grpc/xds/AbstractXdsClient.java

```diff
@@ -180,24 +172,10 @@ void adjustResourceSubscription(ResourceType type) {
    * and sends an ACK request to the management server.
    */
   // Must be synchronized.
-  void ackResponse(ResourceType type, String versionInfo, String nonce) {
-    switch (type) {
-      case LDS:
-        ldsVersion = versionInfo;
-        break;
-      case RDS:
-        rdsVersion = versionInfo;
-        break;
-      case CDS:
-        cdsVersion = versionInfo;
-        break;
-      case EDS:
-        edsVersion = versionInfo;
-        break;
-      case UNKNOWN:
-      default:
-        throw new AssertionError("Unknown resource type: " + type);
-    }
+  <T extends ResourceUpdate> void ackResponse(XdsResourceType<T> xdsResourceType,
```

Comment:
I wouldn't expect `T` to be defined here, as it is unused and likely not necessary in the future. It is "okay" but `?` is more ideal.